label_studio/data_import/uploader.py

```diff
@@ -145,33 +145,95 @@ def tasks_from_url(file_upload_ids, project, request, url, could_be_tasks_lists)
                 file_content = file.read()
                 if isinstance(file_content, str):
                     file_content = file_content.encode()
-        file_upload = create_file_upload(request, project, SimpleUploadedFile(filename, file_content))
+        file_upload = create_file_upload(user, project, SimpleUploadedFile(filename, file_content))
         if flag_set('fflag_fix_back_lsdv_4568_import_csv_links_03032023_short') and file_upload.format_could_be_tasks_list:
-            could_be_tasks_lists = True
+            could_be_task_list = True
         file_upload_ids.append(file_upload.id)
         tasks, found_formats, data_keys = FileUpload.load_tasks_from_uploaded_files(project, file_upload_ids)
 
     except ValidationError as e:
         raise e
     except Exception as e:
         raise ValidationError(str(e))
-    return data_keys, found_formats, tasks, file_upload_ids, could_be_tasks_lists
+    return data_keys, found_formats, tasks, file_upload_ids, could_be_task_list
+
+
+def create_file_uploads(user, project, FILES):
+    could_be_task_list = False
+    file_upload_ids = []
+    check_request_files_size(FILES)
+    check_extensions(FILES)
+    for _, file in FILES.items():
+        file_upload = create_file_upload(user, project, file)
+        if file_upload.format_could_be_tasks_list:
+            could_be_task_list = True
+        file_upload_ids.append(file_upload.id)
+
+    logger.debug(f'created file uploads: {file_upload_ids} could_be_task_list: {could_be_task_list}')
+    return file_upload_ids, could_be_task_list
+
+
+def load_tasks_for_async_import(project_import, user):
+    """ Load tasks from different types of request.data / request.files saved in project_import model
+    """
+    file_upload_ids, found_formats, data_keys = [], [], set()
+    could_be_task_list = False
+
+    if project_import.file_upload_ids:
+        file_upload_ids = project_import.file_upload_ids
+        tasks, found_formats, data_keys = FileUpload.load_tasks_from_uploaded_files(project_import.project, file_upload_ids)
+
+    # take tasks from url address
+    elif project_import.url:
+        url = project_import.url
+        # try to load json with task or tasks from url as string
+        json_data = str_to_json(url)
+        if json_data:
+            file_upload = create_file_upload(request.user, project_import.project, SimpleUploadedFile('inplace.json', url.encode()))
```

Comment:
```suggestion
            file_upload = create_file_upload(user, project_import.project, SimpleUploadedFile('inplace.json', url.encode()))
```

request is no longer provided in context, need to update this to user.